Only change the line height of the view line

remcohaszing opened the pull request:
Previously, if a decoration specified a custom line height, this affected all viewlines that are covered by model line. Now the line height is only applied to the view lines that render the decoration.

It is still possible to change the line height of an entire view line by setting the decoration option `isWholeLine`.

<img width="721" height="801" alt="image" src="https://github.com/user-attachments/assets/f98dbaf9-bc80-47bc-ac7e-4d4e3c4d7b0e" />

Closes #249967
Closes #295179

cc @aiday-mar

aiday-mar commented:
Hi @remcohaszing thank you for this PR. I am not sure we should apply the line height only to the view line because as you mentioned if line wrapping is turned on, and you change the size of the window, then the decorations will change position and you would have to change the size of the line heights. This means the line heights would dynamically change as you resize the screen. This adds a whole layer of complexity to the work. Could we start off by opening an issue about this, and discussing it within the issue, how important of an issue this is, if its urgent, before writing this code and merging it?

remcohaszing committed:
Only change the line height of the view line

Previously, if a decoration specified a custom line height, this
affected all viewlines that are covered by model line. Now the line
height is only applied to the view lines that render the decoration.

It is still possible to change the line height of an entire view line by
setting the decoration option `isWholeLine`.
